rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -1,72 +1,165 @@
 
-        @asynccontextmanager
-        async def lifespan(app: FastAPI):
-            await db.connect()
-            yield
-            await db.disconnect()
+    @asynccontextmanager
+    async def lifespan(self, _: FastAPI):
+        await db.connect()
+        yield
+        await db.disconnect()
 
-        self.app = FastAPI(
+    def run(self):
+        app = FastAPI(
             title="AutoGPT Agent Server",
             description=(
                 "This server is used to execute agents that are created by the "
                 "AutoGPT system."
             ),
             summary="AutoGPT Agent Server",
             version="0.1",
-            lifespan=lifespan,
+            lifespan=self.lifespan,
         )
-        self.execution_queue = queue
 
         # Define the API routes
-        self.router = APIRouter()
-        self.router.add_api_route(
+        router = APIRouter()
+        router.add_api_route(
+            path="/blocks",
+            endpoint=AgentServer.get_agent_blocks,
+            methods=["GET"],
+        )
+        router.add_api_route(
+            path="/agents",
+            endpoint=AgentServer.get_agents,
+            methods=["GET"],
+        )
+        router.add_api_route(
+            path="/agents/{agent_id}",
+            endpoint=AgentServer.get_agent,
+            methods=["GET"],
+        )
+        router.add_api_route(
+            path="/agents",
+            endpoint=AgentServer.create_agent,
+            methods=["POST"],
+        )
+        router.add_api_route(
             path="/agents/{agent_id}/execute",
-            endpoint=self.execute_agent,
+            endpoint=AgentServer.execute_agent,
+            methods=["POST"],
+        )
+        router.add_api_route(
+            path="/agents/{agent_id}/executions/{run_id}",
+            endpoint=AgentServer.get_executions,
+            methods=["GET"],
+        )
+        router.add_api_route(
+            path="/agents/{agent_id}/schedules",
+            endpoint=AgentServer.schedule_agent,
             methods=["POST"],
         )
-        self.app.include_router(self.router)
+        router.add_api_route(
+            path="/agents/{agent_id}/schedules",
+            endpoint=AgentServer.get_execution_schedules,
+            methods=["GET"],
+        )
+
+        app.include_router(router)
+        uvicorn.run(app, host="0.0.0.0", port=8000)
 
-    async def execute_agent(self, agent_id: str, node_input: dict):
+    @staticmethod
+    async def get_agent_blocks() -> list[dict]:
+        return [v.to_dict() for v in await block.get_blocks()]
+
+    @staticmethod
+    async def get_agents() -> list[str]:
+        return await graph.get_graphs()
+
+    @staticmethod
+    async def get_agent(agent_id: str) -> graph.Graph:
+        agent = await graph.get_graph(agent_id)
+        if not agent:
+            raise HTTPException(status_code=404, detail=f"Agent #{agent_id} not found.")
+
+        return agent
```

Comment:
Here Agent == Graph.
I want to avoid the notion of Graph & Node for the user, so there should not be any mention of that on the API layer.
And there should not be any mention of agent at the implementation layer.